core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts

```diff
@@ -17,100 +17,126 @@
-                function(customizationArgsDict) {
-                  for (var arg in customizationArgsDict) {
-                    if (customizationArgsDict.hasOwnProperty(arg)) {
-                      that.setData(arg, customizationArgsDict[arg]);
+                rteHelperService.openCustomizationModal(
+                  customizationArgSpecs,
+                  customizationArgs,
+                  function(customizationArgsDict) {
+                    for (var arg in customizationArgsDict) {
+                      if (customizationArgsDict.hasOwnProperty(arg)) {
+                        that.setData(arg, customizationArgsDict[arg]);
+                      }
+                    }
+                    /**
+                    * This checks whether the widget has already been inited
+                    * and set up before (if we are editing a widget that
+                    * has already been inserted into the RTE, we do not
+                    * need to finalizeCreation again).
+                    */
+                    if (!that.isReady()) {
+                      // Actually create the widget, if we have not already.
+                      editor.widgets.finalizeCreation(container);
                     }
-                  }
-                  /**
-                  * This checks whether the widget has already been inited
-                  * and set up before (if we are editing a widget that
-                  * has already been inserted into the RTE, we do not
-                  * need to finalizeCreation again).
-                  */
-                  if (!that.isReady()) {
-                    // Actually create the widget, if we have not already.
-                    editor.widgets.finalizeCreation(container);
-                  }
 
-                  /**
-                   * Need to manually $compile so the directive renders.
-                   * Note that.element.$ is the native DOM object
-                   * represented by that.element. See:
-                   * http://docs.ckeditor.com/#!/api/CKEDITOR.dom.element
-                   */
-                  $compile($(that.element.$).contents())($rootScope);
-                  // $timeout ensures we do not take the undo snapshot until
-                  // after angular finishes its changes to the component tags.
-                  $timeout(function() {
+                    /**
+                     * Need to manually $compile so the directive renders.
+                     * Note that.element.$ is the native DOM object
+                     * represented by that.element. See:
+                     * http://docs.ckeditor.com/#!/api/CKEDITOR.dom.element
+                     */
+                    // console.log(that.element.$);
+                    // $compile($(that.element.$).contents())($rootScope);
+                    // $timeout ensures we do not take the undo snapshot until
+                    // after angular finishes its changes to the component tags.
```

Comment:
Is this comment still relevant? Since the timeout is a couple of lines below now.

Reply:
Not relevant, removed!